lib/ansible/plugins/lookup/onepassword_doc.py

```diff
@@ -0,0 +1,125 @@
+      master_password:
+        description: The password used to unlock the specified vault.
+        default: None
+        aliases: ['vault_password']
+      subdomain:
+        description: The 1Password subdomain to authenticate against.
+        default: None
+      username:
+        description: The username used to sign in.
+      secret_key:
+        description: The secret key used when performing an initial sign in.
+      vault:
+        description: Vault containing the item to retrieve (case-insensitive). If absent will search all vaults
+        default: None
+    notes:
+      - This lookup will use an existing 1Password session if one exists. If not, and you have already
+        performed an initial sign in (meaning C(~/.op/config) exists), then only the C(master_password) is required.
+        You may optionally specify C(subdomain) in this scenario, otherwise the last used subdomain will be used by C(op).
+      - This lookup can perform an initial login by providing C(subdomain), C(username), C(secret_key) and C(master_password).
+      - Due to the B(very) sensitive nature of these credentials, it is B(highly) recommended that you only pass in the minimal credentials
+        needed at any given time. Also, store these credentials in an Ansible Vault using a key that is equal to or greater in strength
+        to the 1Password master password.
+      - This lookup stores potentially sensitive data from 1Password as Ansible facts.
+        Facts are subject to caching if enabled, which means this data could be stored in clear text
+        on disk or in a database.
+      - Tested with C(op) version 0.5.3
+"""
+
+EXAMPLES = """
+# These examples only work when already signed in to 1Password
+- name: Retrieve document contents for MyAwesomeDocument when already signed in to 1Password
+  debug:
+    var: lookup('onepassword_doc', 'MyAwesomeDocument')
+
+- name: Retrieve document contents for MySecureTLSCert from a specific vault
+  debug:
+    var: lookup('onepassword_doc', 'MySecureTLSCert', vault='MyVault')
+
+- name: Retrieve MySecureTLSCert document when not signed in to 1Password
+  debug:
+    var: lookup('onepassword_doc'
+                'MySecureTLSCert'
+                subdomain='MySubdomain'
+                master_password=vault_master_password)
+
+- name: Retrieve MySecureTLSCert document when never signed in to 1Password
+  debug:
+    var: lookup('onepassword_doc'
+                'MySecureTLSCert'
+                subdomain='MySubdomain'
+                master_password=vault_master_password
+                username='tweety@acme.com'
+                secret_key=vault_secret_key)
+"""
+
+RETURN = """
+  _raw:
+    description: document contents requested
+"""
+
+from ansible.plugins.lookup.onepassword import OnePass
+from ansible.plugins.lookup import LookupBase
+from ansible.module_utils._text import to_bytes
+
+
+class OnePassDoc(OnePass):
+
+    def get_document_raw(self, doc_id, vault=None):
```

Comment:
Use this method directly rather than calling it from `get_document_raw()`
```suggestion
    def get_document(self, doc_id, vault=None):
```